db/snapshot/backup_task.cc

```diff
@@ -112,39 +116,82 @@ future<> backup_task_impl::do_backup() {
     gate uploads;
     auto snapshot_dir_lister = directory_lister(_snapshot_dir, lister::dir_entry_types::of<directory_entry_type::regular>());
 
-    for (;;) {
-        std::optional<directory_entry> component_ent;
-        try {
-            component_ent = co_await snapshot_dir_lister.get();
-        } catch (...) {
-            if (!ex) {
-                ex = std::current_exception();
+    using comps_vector = utils::small_vector<std::string, sstables::num_component_types>;
+    using comps_map = std::unordered_map<sstables::generation_type, comps_vector>;
+    comps_map sstable_comps;
+    size_t num_sstable_comps = 0;
+    comps_vector non_sstable_files;
+
+    try {
+        snap_log.debug("backup_task: listing {}", _snapshot_dir.native());
+        for (;;) {
```

Comment:
Note: more nice to write `while (auto component_ent = co_await snapshot_dir_lister.get()) `, but I see it's an existing problem.

Reply:
done in v2 ([06da3f3](https://github.com/scylladb/scylladb/commit/06da3f30f0c2da5e684fbff7b517e97d70d70559))